pkg/evm/evm.go

```diff
@@ -264,21 +264,31 @@ func FundAddress(
 	return nil
 }
 
-func GetSignedTxToMethodWithWarpMessage(
+func GetTxToMethodWithWarpMessage(
 	client ethclient.Client,
+	generateRawTxOnly bool,
+	from common.Address,
 	privateKeyStr string,
 	warpMessage *avalancheWarp.Message,
 	contract common.Address,
 	callData []byte,
 	value *big.Int,
 ) (*types.Transaction, error) {
 	const defaultGasLimit = 2_000_000
-	privateKey, err := crypto.HexToECDSA(privateKeyStr)
-	if err != nil {
-		return nil, err
+	var (
+		privateKey *ecdsa.PrivateKey
+		err        error
+	)
+	if privateKeyStr != "" {
+		privateKey, err = crypto.HexToECDSA(privateKeyStr)
+		if err != nil {
+			return nil, err
+		}
 	}
-	address := crypto.PubkeyToAddress(privateKey.PublicKey)
-	gasFeeCap, gasTipCap, nonce, err := CalculateTxParams(client, address.Hex())
+	if from == (common.Address{}) {
+		from = crypto.PubkeyToAddress(privateKey.PublicKey)
+	}
```

Comment:
Seems that either privateKeyStr == ""  or from == "", maybe we can just do if else here to prevent unintended overriding of `from` values